src/data-validation.jl

```diff
@@ -187,3 +188,109 @@ function _validate_group_consistency!(connection)
+end
+
+function _validate_simple_method_has_only_matching_years!(error_messages, connection)
+    # Validate that the data should have milestone year = commission year
+    # Error otherwise and point out the unmatched rows
+    # - For assets
+    for row in DuckDB.query(
+        connection,
+        "SELECT asset.asset, asset_both.milestone_year, asset_both.commission_year, asset.investment_method
+        FROM asset_both
+        LEFT JOIN asset
+            ON asset.asset = asset_both.asset
+        WHERE asset_both.milestone_year != asset_both.commission_year
+            AND asset.investment_method in ('simple', 'none')
+        ",
+    )
+        push!(
+            error_messages,
+            "Unexpected (asset='$(row.asset)', milestone_year=$(row.milestone_year), commission_year=$(row.commission_year)) in 'asset_both' for an asset='$(row.asset)' with investment_method='$(row.investment_method)'. For this investment method, rows in 'asset_both' should have milestone_year=commission_year.",
+        )
+    end
+
+    # - For flows
+    for row in DuckDB.query(
+        connection,
+        "SELECT flow.from_asset, flow.to_asset, flow_both.milestone_year, flow_both.commission_year,
+        FROM flow
+        LEFT JOIN flow_both
+            ON flow.is_transport
+            AND flow.from_asset = flow_both.from_asset
+            AND flow.to_asset = flow_both.to_asset
+        WHERE flow_both.milestone_year != flow_both.commission_year
+        ",
+    )
+        push!(
+            error_messages,
+            "Unexpected (from_asset='$(row.from_asset)', to_asset='$(row.to_asset)', milestone_year=$(row.milestone_year), commission_year=$(row.commission_year)) in 'flow_both' for an flow=('$(row.from_asset)', '$(row.to_asset)') with default investment_method='simple/none'. For this investment method, rows in 'flow_both' should have milestone_year=commission_year.",
+        )
+    end
+
+    return error_messages
+end
+
+function _validate_simple_method_all_milestone_years_are_covered!(error_messages, connection)
+    # Validate that the data contains all milestone years where milestone year = commission year
+    # Error otherwise and point out the missing milestone years
+    # - For assets
+    for row in DuckDB.query(
+        connection,
+        "WITH asset_simple AS (
+            SELECT asset.asset, year_data.year AS milestone_year, asset.investment_method
+            FROM asset
+            CROSS JOIN year_data
+            WHERE asset.investment_method in ('simple', 'none')
+                AND year_data.is_milestone
+        )
+        SELECT asset_simple.asset, asset_simple.milestone_year, asset_simple.investment_method
+        FROM asset_simple
+        LEFT JOIN asset_both
+            ON asset_simple.asset = asset_both.asset
+            AND asset_simple.milestone_year = asset_both.milestone_year
+            AND asset_simple.milestone_year = asset_both.commission_year
+        WHERE asset_both.commission_year IS NULL
```

Comment:
```suggestion
        "SELECT asset_milestone.asset, asset_milestone.milestone_year, asset.investment_method
        FROM asset_milestone
        LEFT JOIN asset
            ON asset_milestone.asset = asset.asset
        LEFT JOIN asset_both
            ON asset_milestone.asset = asset_both.asset
            AND asset_milestone.milestone_year = asset_both.milestone_year
            AND asset_milestone.milestone_year = asset_both.commission_year
        WHERE asset_both.commission_year IS NULL
            AND asset.investment_method in ('simple', 'none')
```

Following our discussion, this avoids using `year_data` and trusts that `asset_milestone` is defined for every `asset` and `milestone_year`.

Reply:
I agree that this is fine to assume that, because I guess in one way or another we would validation that in the future. So it simplifies the code here.